### ISSUE TITLE
Black Areas after Baking Sectors

### ISSUE THREAD

**Issue by Prophantii** [guycalledfrank/bakery-issues#120]
When Baking a sector, I get big black patches on the ground. I tried to reposition the capture points but the black patches just appear on different positions. I doubt it's shadows from other objects, since you can see the correct shadows in between the black areas. I'm rendering a small building with interior on 100 Texels per Unit, RTX mode, 3 bounces. The building is rather small with just two small rooms. There are no terrains or anything in the sector itself, but some terrains in the scene, if that's important. The Sector has 6 Capture Points and is set on "Capture in Place".  (Not quite sure how many capture points are necessary but the patches appear when baking with a different amount of capture points.) 

 

Example 1
![screenshot1](https://user-images.githubusercontent.com/24207076/128694486-a3436f1b-81ef-4922-a247-0cfc93962468.png)

Example 2
![screenshot3](https://user-images.githubusercontent.com/24207076/128694496-df5f0548-3aee-429c-bf18-4ad697fdc656.png)

Position of Capture Points
![screenshot2](https://user-images.githubusercontent.com/24207076/128694494-46f84783-fec1-4125-b36d-3e124c79e419.png)

![screenshot4](https://user-images.githubusercontent.com/24207076/128695082-339b9bbd-610d-415b-b912-6569d9dc7bca.png)
![screenshot5](https://user-images.githubusercontent.com/24207076/128695085-ed601e94-a9a3-48db-b0d8-0b75791acbf3.png)

**Comment by guycalledfrank** [comment]
Hmmm... how many sectors are there and how are objects distributed among them? Such patches can appear if objects around the floor mesh don't belong to the same sector.

**Comment by Prophantii** [comment]
Thank you for your answer. Currently there is only one single sector, since I was only testing out the sector function. The scene is actually a big outdoor scene. The sector is only one building containing all it's interior (furniture etc). But beneath the building at around floor level is a Terrain and another Ground-Object which is being used as the floor outside of the building. That might just be it. I will test it out and post here if it worked.

**Comment by guycalledfrank** [comment]
Interesting... can you show me sector geometry preview? Is there something intersecting the floor?

![image](https://user-images.githubusercontent.com/7008423/128859970-b585eac2-0a0e-4426-8f6d-1eb94dda3cde.png)

**Comment by Prophantii** [comment]
Just the preview button right? My PC doesn't seem to be able to handle that. Unity freezes when pressing it or rather it doesn't seem to process anything. 
![grafik](https://user-images.githubusercontent.com/24207076/129013754-e01e0f14-35e1-4a86-a5d6-6e1aeb6a8a83.png)

After disabling this Groundmesh though, the bake of the sector went perfectly fine. There are no more black patches to be seen. 

![grafik](https://user-images.githubusercontent.com/24207076/129014037-c82f4620-a170-4182-bc5c-4b4d9e916fd2.png)

**Comment by guycalledfrank** [comment]
>My PC doesn't seem to be able to handle that. Unity freezes when pressing it or rather it doesn't seem to process anything.

Weird. It should work. Does it work in my example scene? Do you have to kill Unity process after it freezes?

Can you check it with Terrain Optimization disabled? It looks like it doesn't work right when enabled.

**Comment by Prophantii** [comment]
Yes I need to kill the Unity process. I tried it with Terrain Optimization disabled but it still freezes. I'm a little dumbfounded. Like I said, when disabling the ground layers the shadows are fine (for the most part). Some Objects don't throw a shadow upon them selves. It's like they only receive indirect light or something. Probably need to test out more options.

**Comment by guycalledfrank** [comment]
If there is any test package/project you can send to debug, it would help a lot. I assume it would only need the terrain and some walls to test it.
